docker/Dockerfile.genvm

```diff
@@ -1,29 +1,26 @@
-FROM python:3.12.2-slim as base
-
+FROM python:3.12.4-slim as base
 ARG path=/app
-  
-RUN mkdir $path
+RUN addgroup --system genvm-group && adduser --system --ingroup genvm-group genvm-user
+RUN mkdir -p $path && chown -R genvm-user:genvm-group $path
 WORKDIR $path
-COPY ../.env .
-ADD genvm $path/genvm
-ADD common $path/common
-RUN ln genvm/base/llms.py /tmp/llms.py
-RUN ln genvm/base/context_wrapper.py /tmp/context_wrapper.py
-RUN ln genvm/base/contract_runner.py /tmp/contract_runner.py
-RUN ln genvm/base/equivalence_principle.py /tmp/equivalence_principle.py
-RUN pip install --no-cache-dir -r genvm/requirements.txt
-RUN pip install Flask
-
-ENV PYTHONPATH "${PYTHONPATH}:/${path}"
+COPY --chown=genvm-user:genvm-group ../.env .
+COPY --chown=genvm-user:genvm-group genvm $path/genvm
+COPY --chown=genvm-user:genvm-group common $path/common
+RUN ln -s genvm/base/llms.py /tmp/llms.py && \
+    ln -s genvm/base/context_wrapper.py /tmp/context_wrapper.py && \
+    ln -s genvm/base/contract_runner.py /tmp/contract_runner.py && \
+    ln -s genvm/base/equivalence_principle.py /tmp/equivalence_principle.py && \
+    pip install --no-cache-dir -r genvm/requirements.txt && \
+    pip install Flask
+USER genvm-user
+ENV PYTHONPATH="${PYTHONPATH}:/${path}"
 ENV FLASK_APP=genvm/server.py
 
 ###########START NEW IMAGE : DEBUGGER ###################
 FROM base as debug
 RUN pip install debugpy
-
-CMD python -m debugpy --listen 0.0.0.0:${GENVMDEBUGPORT} -m flask run -h 0.0.0.0 -p ${FLASK_SERVER_PORT}
+CMD ["python", "-m", "debugpy", "--listen", "0.0.0.0:${GENVMDEBUGPORT}", "-m", "flask", "run", "-h", "0.0.0.0", "-p", "${FLASK_SERVER_PORT}"]
 
 ###########START NEW IMAGE: PRODUCTION ###################
 FROM base as prod
-
-CMD flask run -h 0.0.0.0 -p ${FLASK_SERVER_PORT}
\ No newline at end of file
+CMD ["flask", "run", "-h", "0.0.0.0", "-p", "${FLASK_SERVER_PORT}"]
```

Comment:
@YeagerAI-Bulat have you checked that this continues to work as expected? I remember there are some differences between the shell and exec forms when handling envvars

https://docs.docker.com/reference/dockerfile/#variable-substitution

Reply:
I have changed it